src/cli/serve/integration_tests/serverless_config_flag.test.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { spawnSync } from 'child_process';
+
+import { REPO_ROOT } from '@kbn/repo-info';
+
+interface LogEntry {
+  message: string;
+  log: {
+    level: string;
+  };
+}
+
+const consoleJsonLog = JSON.stringify({
+  root: { level: 'fatal', appenders: ['console-json'] },
+  appenders: { 'console-json': { type: 'console', layout: { type: 'json' } } },
+});
+
+describe('cli serverless project type', () => {
+  it(
```

Comment:
Done in [`ed7b220` (#155754)](https://github.com/elastic/kibana/pull/155754/commits/ed7b22035217c0f1addcc5ca27681de84b1c616b). I added unit tests because that way I could mock the underlying services. I noticed the _happy-path_ integration tests are skipped, so I assume they are flaky. I don't want to introduce more flakiness.